File: q6.py
import numpy as np
class Point:
    def __init__(self,x_init,y_init):
        self.x = x_init
        self.y = y_init

# check if point lies on the curve
def isPoint(a1):
    y=a1.y*a1.y
    y=np.mod(y,p)
    x=(a1.x*a1.x*a1.x + a*a1.x + b)
    x=np.mod(x,p)
    if(x==y):
        return True
    else:
        return False

# print all point on the curve
def allPoints(p,a,b):
    for i in range(0, p):
        for j in range(0,p):
            a2=Point(i,j)
            if(isPoint(a2)==True):
                print((i,j))

if __name__ == "__main__":
    a = 0
    b = -2
    p = 5
    print("All Points on the curve are :")
    print(allPoints(p,a,b))

File: q6.c
#include <stdio.h>

int main() {
    int x, y;

    printf("Points on elliptic curve E: y^3 ≡ x^3 - 2x (mod 5)\n");
    for (x = 0; x < 5; x++) {
        for (y = 0; y < 5; y++) {
            if ((y*y*y) % 5 == ((x*x*x) - 2*x) % 5) {
                printf("(%d, %d)\n", x, y);
            }
        }
    }

    return 0;
}
